cinema/views.py

```diff
@@ -97,12 +100,26 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
+        elif self.action == "upload_image":
+            return MovieImageSerializer
 
         return MovieSerializer
 
+    @action(
+        methods=["post"],
+        detail=True,
+        url_path="upload-image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Use serializer.is_valid(raise_exception=True) to simplify code. When you use raise_exception=True, if the serializer is invalid, an exception (serializers.ValidationError) is raised. DRF catches this exception and automatically returns a 400 Bad Request response with the appropriate error messages.